source/renderer/app/utils/transaction.js

```diff
@@ -0,0 +1,275 @@
+// @flow
+import React from 'react';
+import moment from 'moment';
+import BigNumber from 'bignumber.js';
+import {
+  WalletTransaction,
+  TransactionTypes,
+} from '../domains/WalletTransaction';
+import { formattedWalletAmount } from './formatters';
+import type {
+  DateRangeType,
+  TransactionFilterOptionsType,
+} from '../stores/TransactionsStore';
+import { DateRangeTypes } from '../stores/TransactionsStore';
+
+const AMOUNT_RAW_LENGTH_LIMIT = 10;
+
+export const generateFilterOptions = (
+  transactions: Array<WalletTransaction>
+) => {
+  const dates = transactions
+    .filter(({ date }) => !!date)
+    .map(({ date }) => (date ? date.getTime() : 0));
+  const amounts = transactions.map(({ amount }) => Math.abs(amount.toNumber()));
+  const dateRange = DateRangeTypes.ALL;
+  const fromDate =
+    dates.length > 0 ? moment(Math.min(...dates)).format('YYYY-MM-DD') : '';
+  const toDate =
+    dates.length > 0 ? moment(Math.max(...dates)).format('YYYY-MM-DD') : '';
+  const fromAmount = amounts.length > 0 ? Math.min(...amounts).toString() : '';
+  const toAmount = amounts.length > 0 ? Math.max(...amounts).toString() : '';
+  const incomingChecked = true;
```

Comment:
In case the user disables both incoming and outcoming, we should check the first unchecked option.